doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,54 @@
+"""
+===============================================
+Reconstruct dust-covered image using inpainting
+===============================================
+
+It is possible that dust gets accumulated on the reference mirror and causes
+dark spots to appear on direct images. This example reproduces the steps taken
+to perform OCT (Optical Coherence Tomography [1]_) dust removal in images.
+This application was first discussed by Jules Scholler in [2]_.
+
+.. [1] Vinay A. Shah M.D. (2015)
+       `Optical Coherence Tomography <https://eyewiki.aao.org/Optical_Coherence_Tomography#:~:text=3%20Limitations-,Overview,at%20least%2010%2D15%20microns.>`_,
+       American Academy of Ophthalmology.
+.. [2] Jules Scholler (2019) "Image denoising using inpainting":
+       `<https://www.jscholler.com/2019-02-28-remove-dots/>`_
+
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import imageio.v3 as iio
+import plotly.express as px
+
+
+#################################################################################
+# The dataset that we are using in this example is an image sequence showing
+# the palisades of Vogt in vivo in a human cornea. The file has been
+# acquired in TIFF format.
+
+raw = iio.imread('data/raw.tiff')
+
+print(f'shape: {raw.shape}')
+
+#################################################################################
+# We visualise the image sequence by taking advantage of  the *animation_feature*
+# parameter in Plotly's *imshow* function. We set this feature to 0 to slice the
+# image sequence along the temporal axis.
+
+fig = px.imshow(raw, animation_frame=0, binary_string=True,
+                labels=dict(animation_frame="slice"),
+                height=500, width=500,
+                title="Animated Visualisation")
+fig.show()
+
+#################################################################################
+# In order to create a mask, we use the statistically average image over all the
+# images in the TIFF file. This average frame is computed over time.
```

Comment:
```suggestion
# Unfortunately, many image viewers, such as matplotlib's `imshow`, are only capable 
# of displaying 2D data. We calculate the element-wise sum of the 2D arrays along the 
# frame (axis=0) of the 3D array `image_sequence`, which result in the 2D numpy array
# `image_sequence_mean`.
```
Consider this explanation for why we’re taking the average over time.

Reply:
Thanks for the suggestion! Although, I think that this explanation misses a couple of points, namely:
- What axis are we computing the average over so that it makes sense to us? (The time axis)
- I think that '3D array' is not specific enough, we could also be talking about RGB images with one dimension for color channels
- It misses on the purpose behind creating the mean image, which is to create a thresholding mask. The bit about matplotlib is a bit misleading because we have already visualised the dataset using plotly, so it's not necessary to create a mean image just for those purposes.

Let me know what you think! :)

Reply:
- The image dimensions are provided in the following order: `plane, row, column`. We are computing the average over the plane/frame. i.e each plane/frame represents an image at a time point. 
- The image `raw.tiff` is a volumetric(3D) image i.e 3D grayscale. [See here](https://scikit-image.org/docs/stable/user_guide/numpy_images.html#coordinate-conventions)
- Maybe it doesn't capture all of its purpose, however, the method `threshold_local` accepts `(N, M[, ..., P]) ndarray` grayscale input image as a parameter. Therefore, we are computing the mean to conform with the input parameter.

These are the thoughts that informed suggestion.
Your thoughts? :)

Reply:
> Consider this explanation for why we’re taking the average over time.

Hmm, I don't think that's the explanation. The reason we are averaging over time is that the dust-covered spots are still (don't move over time) whereas the in-vivo tissue (that we are interested in) *does*!